src/Swup.ts

```diff
@@ -332,21 +332,21 @@ export default class Swup {
 		const { url, hash } = Location.fromUrl(href);
 		const animate = this.options.animateHistoryBrowsing;
 		const resetScroll = this.options.animateHistoryBrowsing;
-		this.context = this.createContext({
-			to: url,
-			hash,
-			event,
-			animate,
-			resetScroll,
-			popstate: true
-		});
 
 		// Does this even do anything?
 		// if (!hash) {
 		// 	event.preventDefault();
 		// }
 
 		this.hooks.triggerSync('popState', { event }, () => {
+			this.context = this.createContext({
```

Comment:
Could you briefly explain why the context must be created inside the handler instead of before? Just for me to get a better feel for the concept.

Reply:
To avoid creating context in case the `popState` hook is replaced and the visit is never performed. But now that you mention it, I think it's better to revert this and make sure we're creating the context correctly, even if somebody replaces the hook.